### ISSUE TITLE
[3.9.1] backward compatibility is broken

### ISSUE THREAD

**Issue by alexrififi** [spatie/laravel-data#590]
**✏️ Describe the bug**

I have LocationData with

```php
    public function __construct(
        public int|null|Optional $parent_id,
    ) {
    }
```

**Optional** is present for partial API update as [in the documentation](https://spatie.be/docs/laravel-data/v3/as-a-data-transfer-object/optional-properties)

In version `3.9.0`, if `parent_id` is `null` in the Model, then in the data object `parent_id` is also equal to `null`.

In version `3.9.1`, if `parent_id` is `null` in the Model, then in the data object `parent_id` is absent

**✅ Expected behavior**

If `parent_id` is `null` in the Model, then in the data object `parent_id` also to be equal to `null`. Otherwise, it is not clear how to use `Optional` in this case (when there may be a partial API update, but null in the model also matters)

**🙏 The reason**

It looks like this commit https://github.com/spatie/laravel-data/commit/3b63875338cd6ef29fe362421a4e2186338f5b2d

**🖥️ Versions**

Laravel: 10.28.0
Laravel Data: 3.9.1
PHP: 8.2

**Comment by Ferikl** [comment]
Same issue!

**Comment by rubenvanassche** [comment]
Fixed in 3.9.2
